dotnet/src/SemanticKernel.Core/Search/TextSearchExtensions.cs

```diff
@@ -0,0 +1,285 @@
+        {
+            arguments.TryGetValue("query", out var query);
+            if (string.IsNullOrEmpty(query?.ToString()))
+            {
+                return [];
+            }
+
+            var parameters = function.Metadata.Parameters;
+
+            searchOptions ??= new()
+            {
+                Count = GetArgumentValue(arguments, parameters, "count", 2),
+                Offset = GetArgumentValue(arguments, parameters, "skip", 0)
+            };
+
+            var result = await textSearch.GetSearchResultsAsync(query?.ToString()!, searchOptions, cancellationToken).ConfigureAwait(false);
+            return await result.Results.ToListAsync(cancellationToken).ConfigureAwait(false);
+        }
+
+        options ??= DefaultGetSearchResultsMethodOptions();
+        return KernelFunctionFactory.CreateFromMethod(
+                GetSearchResultAsync,
+                options);
+    }
+    #endregion
+
+    #region private
+    /// <summary>
+    /// Get the argument value from <see cref="KernelArguments"/> or users default value from
+    /// <see cref="KernelReturnParameterMetadata"/> or default to the provided value.
+    /// </summary>
+    /// <param name="arguments">KernelArguments instance.</param>
+    /// <param name="parameters">List of KernelReturnParameterMetadata.</param>
+    /// <param name="name">Name of the argument.</param>
+    /// <param name="defaultValue">Default value of the argument.</param>
+    private static int GetArgumentValue(KernelArguments arguments, IReadOnlyList<KernelParameterMetadata> parameters, string name, int defaultValue)
+    {
+        arguments.TryGetValue(name, out var value);
+        return (value as int?) ?? GetDefaultValue(parameters, "count", defaultValue);
+    }
+
+    /// <summary>
+    /// Get the argument value <see cref="KernelReturnParameterMetadata"/> or default to the provided value.
+    /// </summary>
+    /// <param name="parameters">List of KernelReturnParameterMetadata.</param>
+    /// <param name="name">Name of the argument.</param>
+    /// <param name="defaultValue">Default value of the argument.</param>
+    private static int GetDefaultValue(IReadOnlyList<KernelParameterMetadata> parameters, string name, int defaultValue)
+    {
+        var value = parameters.FirstOrDefault(parameter => parameter.Name == name)?.DefaultValue;
+        return value is int intValue ? intValue : defaultValue;
+    }
+
+    /// <summary>
+    /// Utility method to map a collection of arbitrary search results to a list of strings.
+    /// </summary>
+    /// <param name="resultList">Collection of search results.</param>
+    /// <param name="mapToString">Optional mapper function to convert a search result to a string.</param>
+    /// <returns></returns>
+    private static List<string> MapToStrings(IEnumerable<object> resultList, MapSearchResultToString? mapToString = null)
```

Comment:
There's no need to return a List if it is later upcast to IEnumerable by the calling method. The ToList call can be dropped in this case.